src/integrationTest/groovy/CompilationAndGeneration.groovy

```diff
@@ -0,0 +1,110 @@
+import org.junit.*
+import static org.junit.Assert.assertTrue
+import static org.junit.Assert.fail
+import org.gradle.testfixtures.ProjectBuilder
+import org.gradle.api.Project
+import org.gradle.tooling.GradleConnector
+import org.gradle.tooling.ProjectConnection
+import org.gradle.tooling.BuildLauncher
+import java.util.Map
+import java.util.HashMap
+import cruise.umple.util.SampleFileWriter;
+
+public class CompilationAndGeneration {    
+    private static final DIRECTORIES_ROOT = "src/integrationTest/resources/TestProject/"
+    private static final Map<String, ArrayList<String>> DIRECTORIES_TO_CHECK
+    // Key = path to directory we want to check, value = the files we expect to find in the directory
+    static {
+        Map<String, ArrayList<String>> aMap = new HashMap<String, ArrayList<String>>()
```

Comment:
What's advantages of doing `<String, String>` vs `<String, ArrayList<String>>`? I can see how it's a bit more convenient to simply work with `String`s and let the `Multimap` worry about what collection to prepare for you when you call `get`. Does Guava somehow make a decision about what `Collection<String>` to use based on performance? How does it "handle the data better than you by default"?

Regardless of if we use `<String, String>` or `<String, ArrayList<String>>`, in our case I think it's most convienient to work with `ArrayList<String>` when doing the `put`.

Sure, you could do something like:

```
builder.put(DIRECTORIES_ROOT + "subproj/build/classes/generatedSource", "Subproj.class")
builder.put(DIRECTORIES_ROOT + "subproj/build/classes/generatedSource", "Subproj2.class")
```

And then rely on `Multimap` to create some kind of collection when I call `.get(DIRECTORIES_ROOT + "subproj/build/classes/generatedSource")` But it's fewer lines of code if I add everything at once as an `ArrayList<String>`. And if I do it that way, it makes more sense to me to explicitly specify `<String, ArrayList<String>>` as the type of the `Multimap`.